packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -296,6 +297,12 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
         isSelected = true;
       }
 
+      const indexes: GridRowIndexes = {
+        visibleRows: currentPage.range.firstRowIndex + nextRenderContext.firstRowIndex! + i,
```

Comment:
This value is wrong. I suggested a new formula in https://github.com/mui/mui-x/pull/3519/files/6cfcd930f646a7866b46456fade5d35c0dc31e26#diff-7fb7dbb1d7fe2c699d9fc5cfc5c6a56b2489af3b46dc1a2a8eaa3404575cacbd